core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -327,6 +328,63 @@ public synchronized void losePrimacy() {
     LOG.info("Raft server successfully restarted");
   }
 
+  @Override
+  public synchronized SnapshotPResponse snapshot() {
+    SnapshotPResponse.Builder builder = SnapshotPResponse.newBuilder();
+    mSnapshotAllowed.set(true);
+    try {
+      CopycatClient client = createClient();
+      long sequence = ThreadLocalRandom.current().nextLong(Long.MIN_VALUE, 0);
+      CompletableFuture<Void> future = client.submit(new JournalEntryCommand(
+          JournalEntry.newBuilder().setSequenceNumber(sequence).build()));
+      try {
+        future.get(5, TimeUnit.SECONDS);
+      } catch (TimeoutException | ExecutionException e) {
+        LOG.info("Exception submitting entry: {}", e.toString());
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        return builder.setTriggered(false)
+            .setMessage("Interrupted while submitting entry to trigger snapshot").build();
+      }
+      if (!mStateMachine.isSnapshotting()) {
+        return builder.setTriggered(false)
+            .setMessage("Do not fulfill Raft journal system snapshot requirement").build();
+      }
+
+      waitForSnapshotting(mStateMachine, builder);
+      if (builder.hasSucceed()) {
+        return builder.build();
+      }
+      return builder.setTriggered(true).setSucceed(true).build();
+    } finally {
+      if (mPrimarySelector.getState() == PrimarySelector.State.PRIMARY) {
```

Comment:
this method is synchronized with `gainPrimacy` and `losePrimacy`, so I think it's safe to always set `mSnapshotAllowed` to `false` at the end. If the primary selector switched to secondary while the snapshot was happening, we will call `losePrimacy` afterward, which will set `mSnapshotAllowed` to `true`

Reply:
Make sense, remove the Primary selector state check